\title{Using shotgun metagenomic data for SSU rRNA gene community analysis}
\author{}

\documentclass[12pt]{article}
\usepackage[a4paper, margin=1in]{geometry}
\usepackage[parfill]{parskip}

% set double space
\usepackage{setspace}
\doublespacing

% set times fond
\usepackage{times}

% line number index
\usepackage{lineno}
\linenumbers

% Compile only with pdfLaTeX
\usepackage[pdftex]{graphicx}

% nice table format
\usepackage{booktabs}

% caption adjustment
\usepackage{caption}
\captionsetup[table]{singlelinecheck=off}

\begin{document}
\maketitle

\begin{abstract}

Targeted sequencing of 16s and other short subunit rRNA genes
amplified directly from environmental samples is now standard
procedure for assessing microbial community structures. In contrast,
metagenomic shotgun sequencing does not depend on gene-targeted
primers or PCR amplification and thus is not affected by primer bias
or chimeras. However, it is difficult to detect and recover rRNA genes
from shotgun data and then to analyze them because of the large number
and short lengths of the reads. We present a framework (SSUsearch, or
need a better name; it's not about search anymore; MG-RNA, SG-RNA,
RNAlyze?) to search for short subunit rRNA gene fragments in shotgun
data, followed by supervised classification, clustering, copy
correction and operational taxonomy unit (OTU) based diversity
analysis. Taxonomy profile comparison of shotgun data and amplicon
data from the same sample shows bias against Actinobacteria and
Verrucomicrobia in the V6-V8 primer set and Actinobacteria in the V4
primer. Besides taxonomy based diversity analyses, this pipeline also
enables {\em de novo} OTU based diversity analyses. When applied to
rhizosphere soil samples, shotgun metagenomic data yields higher
diversity estimates than amplicon data but retains the grouping of
samples in ordination analyses. Firmicutes are the most affected by
copy correction in both soil samples. The search model used can
recover Bacteria, Archaea and Eukaryota. This analysis
framework can also be applied to large subunit rRNA, which can
provide additional confirmation of community structure.
\end{abstract}

\section{Introduction}

Microorganisms are distributed in all domains (Virus, Archaea, Bacteria, and
Eukaryota) in the tree of life and recognized for their key roles in ecosystem
processes and they are diverse at the genetic level because of their ancient
history and various habitats \cite{pace1997,ward2002}. Most microorganisms have
not been cultured, which has become especially clear with the advent of
sequencing of SSU rRNA from environmental samples
\cite{ward2002,huse2008,rdp2009}. High-throughput sequencing of PCR-amplified
short hyper-variable regions of the ribosomal Small Subunit (SSU rRNA) gene
provide a means of accessing the uncultured majority \cite{streit2004}.
Although they are dependent on accurate primer sequences, this amplicon
sequencing has been extensively used in community structure surveys and has
extended our understanding of microbial diversity into many environments
\cite{huse2008,caporaso2012miseq}. Meanwhile, the growing size of SSU rRNA gene
databases provide a rich ecological and phylogenetic context for SSU rRNA gene
based community structure surveys \cite{rdp2009,silva2013}.

Unlike gene targeted amplicon sequencing, shotgun sequencing samples from the
entire community, sequencing randomly shared fragments of DNA
\cite{shotgun2004,metahit2010}. While still prone to sequencing error and bias
from environmental DNA extracion, it should therefore provide a more accurate
characterization of microbial diversity \cite{chimeraslayer}.  In particular,
shotgun sequencing provides an improved means to detect divergent sequences not
recovered by standard SSU rRNA gene primers, such as Verrucomicrobia, as well
as eukaryotic members of the community
\cite{baker2003,primereva2008,verruco2011}.

The challenge for using shotgun DNA for rRNA analyses is efficiently searching
for these fragments in large data sets, and the subsequent analysis of the
short reads.  Several methods have been developed for fast SSU rRNA retrieval
in large data sets \cite{ribopicker,metarna,rrnaselector,metaxa}, but none of
them provides further community analysis using the identified rRNA gene
sequences. On the other hand, traditional community analysis tools
\cite{rdp2009,mothur,qiime} are largely designed to handle amplicon sequences
that are amplified by PCR primers. There are two primary types of community
analyses: reference based and OTU based. The reference based method assigns SSU
rRNA gene sequences to bins based on taxonomy of their closest reference
sequences, while OTU based methods assign gene sequences to bins based on de
novo clustering with a specified similarity cutoff (usually 97\%). The
reference based method can be easily applied to shotgun data once SSU rRNA gene
fragments \cite{rdpclassifier}, while the OTU based method relies on methods
that must be adapted to shotgun data.

The main goal of this study is to enable both OTU based and reference
based community analysis of shotgun metagenome data. First, we apply
Hidden Markov Model (HMM) based methods already shown to be fast and
accurate for SSU rRNA search \cite{metarna,rrnaselector,metaxa} using
well-curated and updated training reference sequences from SILVA.
Next, we leverage longer reads assembled from overlapping paired end
reads and take reads mapped to 150bp small hyper-variable reagons of
SSU rRNA genes for {\em de novo} clustering and further diveristy analyses
(@ctb this sentence needs to be fixed; @gjr better now?). This
analysis framework can also be extended to other genes. We also
analyze the ribosomal large subunit (LSU) gene, another phylogenetic
marker on the same rrn operon as the SSU gene, with the same method,
and provides confirmatory analyses. We test the framework on paired
amplicon and shotgun metagenome data from the soil, and examine
several interesting differences between our results.

\section{Materials and Methods}

{\bf Soil sampling, DNA extraction and sequencing.} Rhizosphere soil
was sampled by collecting soil tightly attached (within 2mm) of roots
and bulk soil was sampled obtained from the mid-point between plant
rows. DNA extraction and SSU rRNA gene amplification was previously
described \cite{ed2010}. An SSU rRNA gene primer set (926F:
AAACTYAAAKGAATTGACGG 1392R: ACGGGCGGTGTGTRC) was used to target the
V6-V8 variable region from bacteria, archaea, and eukaryotes in the
standard Joint Genome Institute (JGI) 454 GS FLX and Titanium
sequencing workflow. Another primer set (515F: GTGCCAGCMGCCGCGGTAA
806R: GGACTACHVGGGTWTCTAAT) was used to target the V4 variable region
in the JGI MiSeq sequencing workflow. Shotgun sequencing was done with
Illumina GAII platforms and HiSeq using 250bp insert libraries and
2x150bp reads. One switchgrass bulk soil sample (SB1) was sequenced by
both the 454 Titanium and Illumina GAII platform. Seven replicates of
corn (C1 - C7) and miscanthus (M1 - M7) rhizosphere soil samples were
sequenced the by MiSeq and HiSeq platform.  All these samples were
collected from the GLBRC intensive site at Kellogg Biological Station,
MI.

{\bf Data preprocessing.} The whole analysis framework is shown in
Figure~\ref{fig:flowchart}. Data preprocessing is necessary for both
shotgun and amplicon data due to sequencing error. Trailing bases with
quality score ``2'' encoded by ASCII 66 ``B'' were trimmed off GAII
Illumina 1.5 shotgun data and reads shorter than 30bp and with Ns were
discarded. HiSeq shotgun data were quality trimmed with fast-mcf in
the EA-Utils (http://code.google.com/p/ea-utils) using flags `` -l 50
-q 30 -w 4 -k 0 -x 0 --max-ns 0 -H''. Then paired-end reads
overlapping by more than 10bp were assembled into one long read by
FLASH \cite{flash2011} with ``-m 10 -M 120 -x 0.20 -r 140 -f 250 -s 25
-t 1''. 454 pyrotag amplicon data was processed using the RDP
pyrosequencing pipeline \cite{rdp2009}. Reads are sequenced from the
reverse primer end and extends to forward primer end. Since the
targeted region is about 467bp (926F/1392R), most reads are not long
enough to reach the forward primer. Thus only the reverse primer was
used for quality trimming. The minimum length was set to 400bp and
defaults were used for other parameters.

{\bf Building SSU and LSU rRNA gene models.} First, we quality
trimmed SILVA
\cite{silva2013} SSU and LSU Ref NR database (version 115) sequences
to discard all sequences with ambiguous DNA bases and convert
 ``U'' were converted to ``T''.  We then clustered them at a 97\%
similarity cutoff using pick\_otus.py and pick\_rep\_set.py from
QIIME \cite{qiime}. We next chose the longest representative in each OTU 
further clustered at 80\% similarity cutoff. We collected the longest
sequence in each OTU, resulting in 4027 representative
sequences for the SSU rRNA gene and 1295 for LSU as a phylogenetically
diverse set of reference genes. We further grouped these sequences
into two groups, one with Bacteria and Archaea and the other with
Eukaryota. Each group was used to make a HMM (Hidden Markov
Model) using hmmbuild in HMMER version 3.1 \cite{hmmer3}, and the two HMM
files were concatenated as a single file.

{\bf Identification of SSU and rRNA gene fragments from metagenomic
data.} We searched shotgun data with hmmsearch in HMMER version 3.1
\cite{hmmer3} using the the LSU and SSU HMM models. We used an e-value
of 10 for the newly built HMMs while an e-value of $10^{-5}$ (default)
was used for models in meta-rna \cite{metarna}, when testing the
sensitivity of newly built models. Since hmmsearch does not
automatically search reverse compliment of reads, reverse complements
of all reads were added to the shotgun sequence file before running
hmmsearch. We fed the HMMER hits from the e-value cutoff of 10 into
the multiple sequence aligner (align.seqs) in Mothur
\cite{mothuraligner2009}. For SSU rRNA gene fragments, 18491
full-length SSU rRNA gene sequences (14956 from Bacteria, 2297 from
Archaea, and 1238 from Eukaryota) from the SILVA database
\cite{silva2013} provided in mothur website were used as the template
with flags ``search = suffix, threshold = 0.5, and filp = t''.  For
LSU rRNA, Multiple Sequence Alignment (MSA) of representative
sequences of SILVA LSU Ref NR database clustered at 97\% similarity
cutoff were used as template with the same flags as SSU. Based on
alignment information in report file, those shotgun reads with more
than 50\% of their basepairs mapped to a reference gene were
designated as SSU rRNA or LSU rRNA gene fragments.

{\bf Comparison of taxonomy based microbial community structures
inferred from shotgun and amplicon SSU rRNA gene sequences.} The SSU
rRNA fragments from shotgun data and amplicon data were classified
using RDP's Classifier \cite{rdpclassifier}. The reference SSU rRNA
genes from RDP and SILVA are provided on the Mothur website and were
used as the template, with a bootstrap confidence cutoff for
classification of 50\%. Representative sequences of SILVA LSU Ref NR
clustered at 97\% similarity cutoff were used as a template and the
taxonomy information was extracted from the sequence file for LSU rRNA
gene. The bacterial taxonomy profiles from shotgun data and amplicon
data were compared at phylum level.

{\bf Copy correction.} SSU rRNA gene copy number of each taxon in
Greengenes database have previously been calculated
\cite{copyrighter}. In the taxonomic summary, the abundance of each
taxon is weighted by the inverse of its SSU rRNA gene copy
number. Similarly in OTU based analysis, the abundance of each OTU is
weighted by the inverse of SSU rRNA gene copy number of its taxon.
Unclassified sequences are weighted by the inverse of average copy
number of all taxons in the data.

{\bf Community structure comparison based on OTUs clustered from 150
hyper variable regions.} For the clustering analysis of shotgun and
amplicon data, 150bp in V8 region (E.coli positions: 1227-1377) were
aligned. Reads longer than 100bp were kept in the MSA of V8 region and
then clustered by mcclust with minimum overlap of 25bp
\cite{rdp2009}. The clustering result was converted to the Mothur format
and beta diversity analysis was done by make.shared with
``label=0.03'', dist.shared(calc=thetayc), and pcoa command in
Mothur. When comparing different regions, E. coli positions: 127-277,
577-727, and 997-1147 were chosen for V2, V4, and V6,
respectively. The procrustes analysis provided in QIIME \cite{qiime}
was used. The PCoA results are plotted in the same figure; V2, V4 and
V6 are transformed to minimize the distances between corresponding
points in V8.

{\bf Comparison of OTU based microbial community structures inferred
from shotgun and amplicon SSU rRNA gene sequences.} The abundance of each
OTU in shotgun data and amplicon data (V6-V8 for SB1 and V4 for M1)
from the same DNA extraction were compared to check the consistency
between the two sequencing approaches. Pearson's correlation
coefficient and $R^2$ of linear regression were used to evaluate the
correlation between the two types of data and between 
technical replicates. All the abundances of each OTU were increased by
a pseudocount of one for display in log scale (avoiding zeros). There are
6 technical replicates (lanes) for SB1. We pooled the first three as
SB1\_123 and the others as SB1\_456 to achieve better sequencing
depth.

{\bf Implementation.} An implementation of the pipeline can be found
in https://github.com/jiarong/SSUsearch. Scripts for reproducing on
the figures in this paper can be found @here.

\section{Results}

{\bf Identification of SSU rRNA gene fragments in pair end shotgun
data. } The 2 newly built HMM models (one for Bacteria and Archaea and
one for Eukaryota) cover most hits found by the 3 domain-specific HMM
models (Bacteria, Archaea, and Eukaryota) in meta-rna (\cite{metarna})
in all three test data sets (Table~\ref{tab:hmm_comparison}). We get
longer sequences after merging overlapping pair ends, as seen in the peak
between 150 and 300 bp in Figure~\ref{fig:read_length_dist}. 15566
(0.03\%) of 44,787,632 quality trimmed and pair end merged sequences
in SB1 and 112402 (0.04\%) of 274,060,925 reads in M1 are identified
as SSU rRNA gene fragments.

    \begin{figure}[tbph!]
    \centering
    \includegraphics[width=0.8\textwidth]{figs/read_length_dist.png}

    \caption[Length distribution of identified SSU rRNA gene shotgun
    fragments]{Length distribution of identified SSU rRNA gene shotgun
    fragments. The peak with longer length results from the merged
    pair ends and benefits classification and {\em de novo} clustering
    in downstream analyses.}

    \label{fig:read_length_dist}
    \end{figure}

{\bf Comparison of taxonomy based microbial community structures
inferred from shotgun and amplicon SSU rRNA gene sequences. } The RDP
Classifier can classify short fragments ($>50$bp) of SSU rRNA genes at
the phylum level with high accuracy (@cite RDP classifier). Both shotgun and
amplicon data show Actinobacteria, Proteobacteria and Acidobacteria as
the three most abundant phyla, comprising about 70\% of
community. More sequences are unclassified in the shotgun data than in
the amplicon data. RDP and SILVA show consistent classification
results. However, the 926F/1392R (V6-V8) primer set has bias against
Verrucomicrobia, and the 515F/806R (V4) primer set has bias against
Actinobacteria.

@CTB refer to Figure 2 here, yes??

To take advantage of the fact that shotgun data is untargeted, the
retrieved Large Subunit (LSU) rRNA genes on the same operon (rrn) as
SSU rRNA gene were also classified. Their taxonomy profile is similar
to that of SSU rRNA gene (Pearson's correlation coefficient of 0.87
for SB1 and 0.91 for M1), except that more reads (19.6\%) remain
unclassified. This is expected because of the much lower number of
reference LSU rRNA genes in the SILVA database. The two genes show
consistent community profiles at the Bacterial phylum level and also
confirm the known primer biases. Further, both the LSU and SSU HMMs
show the ability to identify Eukaryotic members and give about
the same taxonomy profile at domain level.

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/V6to8_SB1_taxa}

  \caption[Taxonomy profiles at Bacterial phylum level of shotgun
  fragments and of amplicon reads amplified using V6-V8
  primer]{Taxonomy profiles at Bacterial phylum level of shotgun
  fragments from SSU and LSU rRNA gene and of amplicon reads amplified
  using V6-V8 primer (515F/806R). Classifications are done using both
  RDP and SILVA reference database. Amplicon data shows less
  Actinobacteria in both databases.}

  \label{fig:V6to8_SB1_taxa}
\end{figure}

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/V4_M1_taxa}

  \caption[Taxonomy profiles at Bacterial phylum level of shotgun
  fragments and of amplicon reads amplified using V4 primer]{Taxonomy
  profiles at Bacterial phylum level of shotgun fragments from SSU and
  LSU rRNA gene and of amplicon reads amplified using V4 primer
  (515F/806R). Classifications are done using both RDP and SILVA
  reference database. Amplicon data shows less Actinobacteria in both
  databases.}

  \label{fig:V4_M1_taxa}
\end{figure}

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/LSU_domain_taxa}

  \caption[Taxonomy profile at domain level using SILVA SSU and LSU
  rRNA database]{Supp. Taxonomy profile at domain level using SILVA
  SSU and LSU rRNA database. For both samples (SB1 and M1), SSU and
  LSU show consistent domain level taxonomy distribution (Pearson's
  correlation coefficient = 1).}

  \label{fig:LSU_domain_taxa}
\end{figure}


{\bf Comparison of OTU based microbial community structures inferred
from shotgun and amplicon SSU rRNA gene sequences. }
Figure~\ref{fig:SB1_techrep_OTUscat} shows our analysis pipeline has
good reproducibility between two technical replicates (Pearson's
correlation coefficient = 0.997). Even with log-transformed abundance,
the two replicates have Pearson's correlation coefficient of 0.91 and
linear regression $R^2$ of 0.88. However, OTU abundance in shotgun and
amplicon data does not correlate as well with Pearson's correlation
coefficient of 0.873 in SB1 and 0.581 in M1. There are OTUs with
significant difference in shotgun data and amplicon data.

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/SB1_techrep_OTUscat}

  \caption[Technical reproducibility test]{Supp. Technical
  reproducibility test. X axis shows number of raw reads per OTU in
  log scale in replicate SB1\_123, and y axis shows number of raw
  reads per OUT in log scale in replicate SB1\_456. The size of circle
  is proportional to number of OTUs with the same counts in SB1\_123
  and also in SB1\_456. The counts are in both replicates are
  increased by 1 to avoid 0 counts that can be displaced in log
  scale. There are 6 technical replicates for sample SB1 sequenced in
  6 lanes of Illumina plate using DNA from the same extraction. The
  first three lanes are pooled as SB1\_123 and next three lanes are
  pooled as SB1\_456. The two technical replicates show consistent
  counts for each OTU, and the consistency is higher when the
  abundance of OTUs are higher. Pearson's correlation coefficient is
  0.997 on the original counts.}

  \label{fig:SB1_techrep_OTUscat}
\end{figure}

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/SB1_techrep_OTUscatR2}

  \caption[Progressive dropout analysis using log10 transformed
  data]{Supp. Progressive dropout analysis using log10 transformed
  data. X axis the threshold of OTU abundance and y axis is the $R^2$
  of linear regression of log transformed OTU abundances in two
  replicates when OTUs with lower abundance then threshold are
  discarded. When only OTUs with abundance higher than 25 are
  considered, $R^2$ reaches 0.88, which is acceptable as a balance
  between reproducibility and data loss for low abundance OTUs within
  log10 transformed data.}

  \label{fig:SB1_techrep_OTUscatR2}
\end{figure}

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/SB1_SGvsPT_OTUscat}

  \caption[Comparison of OTU abundance between shotgun and amplicon
  data in sample SB1]{Supp. Comparison of OTU abundance between
  shotgun and amplicon data. X axis shows number of shotgun raw reads
  per OTU in log scale in sample SB1, and y axis shows number of
  amplicon raw reads per OUT in log scale in sample SB1. The size of
  circle is proportional to number of OTUs with the same counts in
  shotgun data and also in amplicon data. The counts are in both
  replicates are increased by 1 to avoid 0 counts that can be
  displaced in log scale. There are OTUs with significant different
  abundance in two types of data (circles deviate from diagonal
  line). Pearson's correlation between two types of data is
  0.873.}  \label{fig:SB1_SGvsPT_OTUscat}
\end{figure}

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/M1_SGvsPT_OTUscat}

  \caption[Comparison of OTU abundance between shotgun and amplicon
  data in sample M1. ]{Supp. Comparison of OTU abundance between
  shotgun and amplicon data in sample M1. X axis shows number of
  shotgun raw reads per OTU in log scale, and y axis shows number of
  amplicon raw reads per OTU in log scale. The size of circle is
  proportional to number of OTUs with the same counts in shotgun data
  and also in amplicon data. The counts in both replicates are
  increased by 1 to avoid 0 counts in log scale. There are OTUs with
  significant different abundance in two types of data (circles
  deviate from diagonal line). Pearson's correlation between the two types
  of data is 0.581.}

  \label{fig:M1_SGvsPT_OTUscat}
\end{figure}

{\bf OTU based diversity analysis of shotgun SSU fragments. }  Shotgun
and amplicon data both show separation of corn and miscanthus samples,
while there are also significant between these two types of data
(Figure ~\ref{fig:V4_SGvsPT_pcoa}). Further, significant separation of
corn and miscanthus samples is also observed when V2, V4, V6, and V8
are used for {\em de novo} OTU clustering in shotgun data (Figure
~\ref{fig:compare_vregion}). Miscanthus samples have higher diversity
than corn samples in all of V2, V4, V6 and V8 regions, though
variations among these regions are also observed (Figure
~\ref{fig:compare_vregion_alpha}).

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/V4_SGvsPT_pcoa}

  \caption[PCoA of OTUs resulting from {\em de novo} clustering with
  shotgun data and amplicon data using 150bp of V4 region]{PCoA of
  OTUs resulting from {\em de novo} clustering with shotgun data and
  amplicon data using 150bp of V4 region. Samples with suffix ``\_PT''
  (teal) are amplicon data and the others (orange) are shotgun
  data. Two types of data both show the separation of corn (circle)
  and miscanthus (square) rhizosphere samples along x-axis while there
  are significant differences between these two types of data shown
  along y-axis (ANOVA p-value $<$ 0.001).}

  \label{fig:V4_SGvsPT_pcoa}
\end{figure}

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/compare_vregion_color_1leg}

  \caption[PCoA analysis of OTUs resulting from {\em de novo}
  clustering with shotgun data using 150bp of V2, V4, V6, V8
  regions]{Supp.?? PCoA analysis of OTUs resulting from {\em de novo}
  clustering with shotgun data using 150bp of V2, V4, V6, V8
  regions. Different variable regions in the shotgun data can be used
  for clustering and show the same grouping of samples, which is
  separation of corn (filled markers) and miscanthus (unfilled
  markers) rhizosphere samples along x-axis. The PCoA results of
  different variable regions are plotted together using procrustes
  analysis in QIIME.}

  \label{fig:compare_vregion}
\end{figure}

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/compare_vregion_alpha.pdf}

  \caption[Alpha diversity of OTUs resulting from {\em de novo}
  clustering with shotgun data using 150bp of V2, V4, V6, V8
  regions]{Alpha diversity of OTUs resulting from {\em de novo}
  clustering with shotgun data using 150bp of V2, V4, V6, V8
  regions. All variable regions show M (miscanthus) is more diverse
  than C (corn), even though there are variation of diversity among
  regions.}

  \label{fig:compare_vregion_alpha}
\end{figure}

{\bf Copy correction. }  Both soil samples (SB1 and M1) show
that Firmicutes and Bacteroidetes have highest fold change after copy
correction. Firmicutes show the highest fold change ($\approx 3$)
(Figure ~\ref{fig:SB1+M1_cc}). Despite the relative abundance
change after copy correction, the clustering of our soils samples
does not change (Figure ~\ref{fig:V4-SG+PT-CC-pcoa}).

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/SB1+M1_cc}

  \caption[Taxonomy summary before and after SSU rRNA gene copy
  correction]{Bacterial phylum level taxonomy summary before and after
  SSU rRNA gene copy correction. Left vertical axis shows percentage
  in total community and right vertical axis shows fold change after
  copy correction. Taxon with relative abundance of more than 0.1\%
  before copy correction are chosen and are ordered based on fold
  change. Subfigure A is for SB1 and B is for M1.}

  \label{fig:SB1+M1_cc}
\end{figure}

\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.8\textwidth]{figs/V4-SG+PT-CC-pcoa}

\caption[PCoA of copy corrected OTUs resulting from {\em de novo} clustering with shotgun data and amplicon data using 150bp of V4 region]{PCoA of copy corrected OTUs resulting from {\em de novo} clustering with shotgun data and amplicon data using 150bp of V4 region. Samples with suffix ``\_PT'' (teal) are amplicon data and the others (orange) are shotgun data. Two types of data both show the separation of corn (circle) and miscanthus (square) rhizosphere samples along x-axis while there are significant differences between these two types of data shown along y-axis (ANOVA p-value $<$ 0.001).}

  \label{fig:V4-SG+PT-CC-pcoa}
\end{figure}


\section{Discussion}
We present, characterize and validate an efficient method for mining
and analyzing SSU rRNA genes from shotgun metagenomic sequence that
has advantages over existing shotgun analysis pipelines such as
MG-RAST. MG-RAST annotates shotgun reads by BLAT search against rRNA
databases and the taxonomy of reads is inferred from the best hit or least
common ancestor of several top hits (\cite{blast,blat,mgrast}).  However,
BLAT or BLAST-like tools are too slow for large datasets and must be
run in parallel on large computer clusters. Moreover, these
pipelines lacks OTU-based community analysis.

HMM based rRNA gene search has been applied extensively
(\cite{metarna,rrnaselector,metaxa}). Here HMM based search is mainly
used as a pre-filter. Thus sensitivity and speed are important and
achieved by using two HMM models (one for bacteria and archaea, and
another for Eukaryota) instead of three, while selecting a
phylogenetically diverse subset from well-curated SILVA reference
sequences to make sure the new HMMs have high sensitivity. SSU rRNA
genes are conserved and the combined model still has high sensitivity
when a relaxed e-value cutoff is used. False positives are tolerable in
this initial search step since there is a following filtering
step.

Our results (Table~\ref{tab:hmm_comparison}) show that our models are
sufficient to recover most of the search results by the models in
meta-rna. It is not necessary to bin SSU rRNA genes into domains at
this step because the search results can be further classified into
domains or even species later, using classifiers. Moreover, since this
analysis focuses on diversity analysis, the search step is optional
and the SSU rRNA fragments identified by other tools can also be fed
directly to the following steps.

An implementation of the Nearest Alignment Space Termination (NAST)
(\cite{mothuraligner2009}) alignment method in
Mothur is used for aligning the SSU fragments and screening false
positives by their identity to the most similar reference in
reference set provided in Mothur. The choice of identity cutoff (50\%)
is arbitrary here, since there is no clear definition of SSU rRNA gene
and those reads with low identity to reference sequences are also
discarded prior to clustering in traditional amplicon analysis pipelines
(\cite{rdp2009, mothur, qiime}). Meanwhile amplicon sequences with
less than 50\% identity to reference sequences are also discarded for
consistency in comparison. An alignment of the SSU rRNA gene fragments
is essential for the later {\em de novo} OTU related
analysis. In comparison to the methods that use only the 16S rRNA gene in
E.coli as alignment template (\cite{kostas2013}), this method takes
advantages of the rich phylogenetic diversity of SSU rRNA genes
provided by SILVA database. Increasing the number of reference
sequences included can improve the quality of alignment, but also
linearly increases the memory required
(\cite{mothuraligner2009,pynast}).

This pipeline also integrates the RDP classifier, which is suitable
for classification of reads as short as 50bp
(\cite{rdpclassifier}). In taxonomy comparison of shotgun and amplicon
data, two main SSU rRNA gene database, RDP and SILVA, are used to make
sure the taxonomy distribution is not biased by the choice of
reference database. In addition, LSU rRNA gene is also used to confirm
the classification by SSU rRNA gene. V6-V8 primer set shows
significant bias against Verrucomicrobia, consistent with other
studies showing that Verrucomicrobia's abundance in soil samples is
underestimated due to primer bias (\cite{verruco2011}). In contrast,
the V4 primer set shows bias towards Verrucomicrobia. This result agrees with
studies showing that V4 primer set has better coverage of
Verrucomicrobia (\cite{verruco2011}). Further, V4 primer sets shows
bias against Actinobacteria. The V4 primer set is reported to cover
92.43\% of Actinobacteria in reference databases, the lowest among
nine common bacterial phyla (Verrucomicrobia, Acidobacteria,
Actinobacteria, Bacteroidetes, Chloroflexi, Firmicutes,
Gemmatimonadetes, Planctomycetes, and Proteobacteria)
(\cite{verruco2011}). Primer bias against Actinobacteria has also been
reported in environmental samples using a Sanger sequencing primer
(24F/1492R) (\cite{actinobias}). The V4 primer set (515F/806R) remains a
good choice as universal primers due to its overall coverage of
bacterial phyla, but attention need to be paid to the missing
Actinobacteria in future studies.

Gene copy number is another source of bias that limits our ability to
accurately profile microbial communities. There are up to 15 SSU rRNA
gene copies in Bacteria and up to 5 in Archaea
(\cite{rrncopy2004}). This pipeline utilizes the CopyRighter SSU rRNA copy database (\cite{copyrighter}). Both of our soil samples
show Firmicutes have highest fold change (Figure
~\ref{fig:SB1+M1_cc}), consistent with the finding in CopyRighter paper
(\cite{copyrighter}). While gene copy correction changes the
absolute and relative abundance of most OTUs or taxa, the grouping pattern of
our soil samples in an ordination analysis is still the same as before
copy correction. Copy correction is still an open problem because
SSU rRNA gene copy number data for most species are lacking and copy
number can be incorrect even for species with complete genome sequences
because of mis-assembly of these repeated regions. It might be OK to
skip copy correction when comparing communities using data targeting
the same gene, because the copy number bias can be treated as
systematic bias. However, copy correction is necessary when different
data are used, e.g. when comparing community profiles using SSU rRNA gene
amplicon data and using annotation from whole genome shotgun
data.

Generally, OTU based analysis provides higher resolution than taxonomy
(\cite{patotuasse2011}) based diversity analysis for community
comparison, mainly due to lack of reference sequence from unculturable
microbes in the databases. The high correlation of OTU abundance in
two technical replicates shows the reproducibility shotgun
data. Moreover, the Pearson's correlation coefficient of log-transformed
OTU abundance in two replicates increased from 0.584 to 0.938 after
OTUs with abundance lower than 25 are discarded. Consistently, $R^2$
of linear regression between log-transformed OTU abundance reaches
0.88 after removing OTUs with abundance lower than 25, which is
comparable to the reproducibility of amplicon data mentioned
in \cite{arabdopsis}. Further, comparison of OTU abundance in shotgun
data and amplicon data sequenced from the same DNA extraction also
show many OTUs have inconsistent abundance in two types of data, which
agrees with the difference seen in the taxonomy based comparison.

The beta-diversity analysis by ordination methods such as PCoA and
NMDS is one of the most common analyses in microbial ecology. To the
best of our knowledge, the methods used in \cite{kostas2013,phylotu}
are the only existing tools that are designed to deal with {\em de
novo} clustering of Illumina shotgun reads. The former could results
in bad alignment by using only the 16S rRNA gene in E.coli as the alignment
template. The latter (PhylOTU) does the OTU clustering of SSU rRNA
gene fragments aligned together over whole gene length, which can be
problematic because fragments aligned to different regions do not
overlap and thus the clustering results are not reliable, even though
the reference sequences included in the clustering process can improve
the results. In our clustering method, we make sure all the sequences
overlap by picking one small region (150bp) and all sequences included
in clustering have lengths longer than 100bp. In addition, we get
longer reads by assembling overlapping pair end reads (Figure
~\ref{fig:read_length_dist}) and these longer reads can make sure more reads
overlap and thus are more suitable for {\em de novo} clustering. Further,
several studies have shown as little as 150bp from a variable region can produce very
similar PCoA results to using 400bp from the same variable regions
(\cite{robknight2007,caporaso2012miseq}). As read lengths increase
with improvement of sequencing technology, longer regions can be
chosen and the clustering results will be even more reliable. Further,
shotgun data give us the flexibility to choose any variable region
(Figure ~\ref{fig:compare_vregion}), and the consistency of results from
different variable region gives us more confidence in the biological
conclusion as well as the method itself.

In this study, the LSU rRNA gene is used mainly as confirmation analysis
for SSU rRNA gene based diversity analysis. However, LSU rRNA gene
offers additional stretches of variable and characteristic sequence
regions due to longer sequence length, and thus yields better phylogenetic
resolution (\cite{lsuprimer}). For the above reason and also the
reason that there are more available references for fungi, the LSU rRNA
gene is commonly used for fungal community studies
(\cite{lsufungal2007,lsuclassifier,lsufungal2012,lsufungal2010}). As a
universal marker gene, LSU rRNA gene is limited by reference sequences
and universal primer sets available (\cite{lsuprimer}). The limitation
of the references is confirmed by more unclassified sequences by LSU
rRNA than SSU rRNA gene (Figure
~\ref{fig:V6to8_SB1_taxa},~\ref{fig:V4_M1_taxa}). This analysis
framework provides a way to avoid the primer limitation and bias for
LSU rRNA gene. The consistency of SSU rRNA gene and LSU rRNA gene
taxonomy profiles at domain and bacteria phylum level of the same
sample shown in this study is promising evidence. With increasing
reference sequences in database and improved curated alignments,
LSU rRNA gene could be as good a marker gene as SSU rRNA gene. Further,
comparison between both genes and across multiple regions can provide
cross validation and thus help us make solid conclusion (Figure
~\ref{fig:compare_vregion}).

\section{Conclusion}

Although shotgun sequencing has advantages -- in particular, no
chimera or primer bias -- analyzing the large data sets has been a big
challenge. In this study, an efficient method for searching and
analyzing SSU rRNA fragments was developed. It takes advantage of
HMMER for searching SSU rRNA gene and LSU rRNA gene from Bacteria,
Archaea and Eukaryota, and can be run on most desktops with more than
4Gb memory (the memory required to load all reference sequences for
alignment). Since SSU rRNA based community analysis is an important
tool in microbial ecology, this method can save projects with existing 
shotgun sequencing from the additional cost of SSU rRNA amplicon
sequencing. As read length and sequencing depth increases, longer and
more SSU rRNA gene fragments can be recovered. Thus clustering and
beta-diversity analysis will become even more reliable. In addition,
we have shown LSU rRNA is very consistent with SSU rRNA gene in terms
of taxonomy. In future, other single copy genes such as {\em rpoB},
{\em gyrB}, and {\em recA} can also be recovered and used for
community structure analysis, as more resolving complements for
current main stream of SSU/LSU rRNA based analysis.

\bibliographystyle{unsrt}
\bibliography{refs.bib}

\section{Supplements}


\begin{figure}[tbph!]
  \centering
  \includegraphics[width=0.95\textwidth]{figs/flowchart}
  \caption[Flowchart of SSUsearch pipeline]{Flowchart of SSUsearch pipeline.}
  \label{fig:flowchart}
\end{figure}


% \usepackage{booktabs}
% \usepackage{caption}
% \captionsetup{singlelinecheck=off}
\begin{table}[tbph!]

 \caption{Comparison of search results using HMM models from SSUsearch
 and meta-rna. Subsets of 5 million reads from a synthetic community,
 M1 and SB1 (soil metagenomes) are used as testing data.}

\label{tab:hmm_comparison}
\begin{tabular}{cccc}
\toprule
     & SSUsearch & meta-rna & overlap \\
\midrule
mock & 6407      & 6458     & 6286 \\
SB1  & 4017      & 3915     & 3889 \\
M1   & 3563      & 3530     & 3510 \\
\bottomrule
\end{tabular}
\end{table}


\end{document}
